### ISSUE TITLE
`<persName>` versus plain `<name>`

### ISSUE THREAD

**Issue by martindholmes** [projectLEMDO/lemdoIssues#2]
Original issue:

We need a policy on the use of `<persName>` versus `<name>`; they're both used in free variation, but `<persName>` with `@ref` does not seem to be processed in running text. We should probably standardize on `<persName>` instead of plain `<name>` where `@ref` points to pers: or pros:, and disallow plain name without `@type`. Update processing appropriately.

Further discussion:

Per meeting discussion 2021-11-29: We should break this problem down into separate areas and deal with each in turn:

1. Find examples of what is not rendering where expected (`<persName>` in running prose not turning into a link???). Fix that.

2. Identify individual contexts where `<persName>` and name are both used (e.g. `<respStmts>`?). Change all `<name>` to `<persName>` and constrain.

3. Finally address the castlist issue. We may just live with that as it is.

**Comment by martindholmes** [comment]
@JanelleJenstad Who's responsible for #1 above? Is this ticket going anywhere or should we just close it?

**Comment by martindholmes** [comment]
@JanelleJenstad Last call on this ticket! If there's nothing specific to work on, I'm going to close it.

**Comment by JanelleJenstad** [comment]
**_We have 10050 `<persName>` elements across the project._** 

**Expected uses of `<persName>`**
- We have 500 `<persName>` elements in PERS1 and 272 in PROS1 (where one would expect to see them).
- `<persName>` also shows up in the edition pages in `<byline>` (probably not necessary because it's effectively running prose, but your processing might be expecting `<persName>`
- and in the character lists inside `<listPerson>` in the header of modern texts.

**Unexpected uses of `<persName>`:**
- In running prose in Performance Intros, annotations, video metadata. 

_**We have 20,669 instances of `<name>`**_

**Expected uses of name:**
- responsibility statements in teiHeaders.
- as child of `<persName>`

So to my mind the only issue is the occasional use of `<persName>` in running prose. If you think that's an inappropriate use of `<persName>`, then we should revisit it and perhaps constrain use with Schematron.

**Comment by martindholmes** [comment]
@JanelleJenstad There are many uses of `<name>` in cases like this:

`This edition would not have been possible without the patient care and good humour of <name ref="pers:GREA1">Brett Hirsch</name>. `

The output in these cases is identical to the use of a `<persName>` in the same context; the name becomes a link which shows the popup bio.

I think I would like to propose that in the long run:

1. All references to people (including in `<respStmts>`) use `<persName>`.
2. All references to orgs (including in `<respStmts>`) use `<orgName>`,
3. Both `<persName>` and `<orgName>` required `@ref` except inside person or org elements. I see no purpose in tagging something as a `<persName>` or an `<orgName>` unless you're linking it to something (although there are 31 cases of such name tagging in our repo right now).
4. The use of `<name>` be eliminated except inside `<persName>` or `<orgName>` where it serves a specific purpose.

What do you think?

**Comment by JanelleJenstad** [comment]
I think this is an excellent solution. Let's make it so. 
I wonder if the instances of `<persName>` and `<orgName>` without `@ref` were created by editors while waiting for us to make an xml:id. We can probably fix those relatively quickly and then write a Schematron rule that those two elements must have the ref attribute. 
I imagine there are other ways to constrain things -- e.g., disallow `<name`> as a child of `<respStmt>`. (Do we do that by changing the content model?)

**Comment by LEMDO-PM** [comment]
@martindholmes, if I go into the files and replace `<name>` with `<persName>` or `<orgName>`, can you and I go in and add Schematron to disallow `<name>` except for as children as those two elements?

**Comment by martindholmes** [comment]
@LEMDO-PM Yes, and I bet you could write the Schematron yourself. Hint: `name[not(parent::persName or parent::orgName)]`.

Try an XPath search for those (prepend // to the above) before committing to the Schematron rule, in case there are edge cases we haven't thought of.

**Comment by LEMDO-PM** [comment]
@martindholmes `<name>` is also a required child of `<respStmt>`. Are we happy with that?

**Comment by martindholmes** [comment]
I don't know why name was used in respStmt as opposed to persName -- perhaps so that orgs and persons could be handled in the same manner? We could change it, although it would be slightly disruptive. Perhaps we should make that a second ticket. @JanelleJenstad what do you think?

**Comment by JanelleJenstad** [comment]
Do we need to handle names in `<respStmt>`s in any special way? If yes, then we should keep `<name>` for that context only, especially if it allows us to process organizations and people the same way in credits. 

If not, then I suggest we deprecate `<name>` and use the more granular `<orgName>`, `<persName>`, and `<placeName>` across the project.

**Comment by martindholmes** [comment]
I'm not sure who the onus is on for this ticket at the moment. I'm going to try changing some respStmts in one file to use persName and orgName and see what happens; if there's any fallout, I can fix it, then we can change them all, as a first step.

**Comment by martindholmes** [comment]
...and I instantly discover that persName is not allowed in respStmt. Have to fix that first.

**Comment by LEMDO-PM** [comment]
That makes sense as a first step to me! That should make the number of `<name>` elements more manageable. I expect that we'll need to deal with the ones in the texts themselves manually as (iirc) many don't have `@ref` yet.

**Comment by martindholmes** [comment]
OK, persName and orgName are now allowed in respStmt, and they are being correctly rendered in the output in the Credits tab, as of rev 12795. We could now mass-convert all the existing respStmts, but before we do that, you should probably update the documentation and examples.

Once that's done, we can convert all the respStmts, and then remove plain `<name>` from the content model of `<respStmt>`.

**Comment by LEMDO-PM** [comment]
Alright, I'll work on that tomorrow and Monday.

**Comment by martindholmes** [comment]
Processing has been added to handle persName in running prose without `@ref`, in rev 13403.

**Comment by martindholmes** [comment]
This seems to be done (MH and JJ).

**Comment by LEMDO-PM** [comment]
@martindholmes @JanelleJenstad Can we add `<objectName>` to our schema? In emdTRIU2_M, we have a ship name that will want italicized and I think `<objectName>` is likely the best element to do that with.

**Comment by martindholmes** [comment]
Fine by me. Do you want to go ahead with it? I think it'll need its @type attribute to be overridden, with a single (so far) value of "ship", because some object names won't be italicized. Once you've added it (assuming it's OK with @JanelleJenstad), you could start looking at XSLT changes we'll need to make.

**Comment by LEMDO-PM** [comment]
Great! I can do that.

**Comment by martindholmes** [comment]
@LEMDO-PM Where are we at with objectName? I think I'm going to close this lengthy ticket and raise a specific one.

**Comment by martindholmes** [comment]
I'm closing this in favour of #193.

**Comment by JanelleJenstad** [comment]
We still need to implement the recommendations @martindholmes made on 2022-11-02
> I think I would like to propose that in the long run:
> 
> 1. All references to people (including in `<respStmts>`) use `<persName>`.
> 2. All references to orgs (including in `<respStmts>`) use `<orgName>`,
> 3. Both `<persName>` and `<orgName>` required `@ref` except inside person or org elements. I see no purpose in tagging something as a `<persName>` or an `<orgName>` unless you're linking it to something (although there are 31 cases of such name tagging in our repo right now).
> 4. The use of `<name>` be eliminated except inside `<persName>` or `<orgName>` where it serves a specific purpose.

We cannot implement these recommendations until the LateX processing can handle `persName`. Once @martindholmes gives the go-ahead, I'll do a global change.

**Comment by martindholmes** [comment]
@JanelleJenstad LaTeX/PDF processing now expects `<persName>`, and will break if `<name>` is encountered. All print XML sources have been changed.

**Comment by martindholmes** [comment]
I'm ready to change all instances of `<name>` for people in `<respStmt>`s to `<persName>`. I Oxygen, find regex:

`<(/?)name`

Restrict to XPath: 

`respStmt[child::name[matches(@ref, '^(pros)|(pers):')]]`

Replace with:

`<$1persName`

Tested on one file. There are over 2,000 instances that will be changed.

**Comment by martindholmes** [comment]
Fix committed for persName in rev 27096.

**Comment by martindholmes** [comment]
We have 645 cases of `<name>` pointing at an org; I think we should fix those too. I'll do that next.

**Comment by martindholmes** [comment]
Fix committed for orgName in rev 27097.

**Comment by JanelleJenstad** [comment]
Thank you!

**Comment by martindholmes** [comment]
I've updated the schema to disallow `<name>` as a child of `<respStmt>`, which should prevent any future instances of this. I'll now close the ticket.

**Comment by martindholmes** [comment]
Reopening because we still have 346 instances of the use of `<name>` outside of `<respStmt>` which are problematic. It's very common in `<sourceDesc>`, for instance. I need to do some testing to ensure that there will be no fallout from changing all those instances, then I'll fix them.

**Comment by martindholmes** [comment]
Committed changes from `<name>` to `<persName>` and `<orgName>` outside of `<respStmt>` in rev 27104.

**Comment by martindholmes** [comment]
I've now removed `@ref` from the `<name>` element, which means all legitimate uses of `<name>` are still OK, but you can't use it in error for `<persName>` or `<orgName>`. Closing this.
